code/game/objects/items/food/_food.dm

```diff
@@ -108,3 +108,10 @@
 	if(!preserved_food)
 		AddComponent(/datum/component/decomposition, mapload, decomp_req_handle, decomp_flags = foodtypes, decomp_result = decomp_type, ant_attracting = ant_attracting)
 
+/obj/item/food/suicide_act(mob/user, show_message=TRUE)
```

Comment:
`/obj/item/food` should not define behavior, it should only be a composite of components/elements. Notice the way `/obj/item/food/Initialize` looks.

This should be its own element for reagent list suicides or similar.